### ISSUE TITLE
Y-Axis labels/tick values are incorrect

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#4859]
The metrics views have Y-Axis label saying "ms" but the tick values I see are, from high to low, [1, 800, 600, 400] and [4, 3.2, 2.4, 1.6, 800]

I assume the "1" is actually in seconds (not milliseconds). So it should show as "1000". Same with the other "low" numbers - those are probably seconds.

We should always show the values in the units shown in the Y-Axis; it is confusing otherwise.

The Metrics tab view:
![image](https://user-images.githubusercontent.com/2029470/160195585-522bd162-d27b-493d-957a-540c19ca6791.png)

and the expanded view:
![image](https://user-images.githubusercontent.com/2029470/160195676-e2ae867f-2341-4efd-8ffb-2d1fa8d0cd26.png)

**Comment by lucasponce** [comment]
> I assume the "1" is actually in seconds (not milliseconds). So it should show as "1000".

Yes, this sounds like some automatic round that could lead to confusion but I'll see what we can do, it's a tradeoff here, I remember some limitations.

Also, note that the "left", "right" y-axis have different metrics, they shouldn't not scale

**Comment by jmazzitelli** [comment]
> Also, note that the "left", "right" y-axis have different metrics, they shouldn't necessarily scale as they represent two different dimensions.

Yes. I agree. This issue was only to point out the single axis values should be in the units that show on the axis label (i.e. if the axis is labeled as "ms", all values should be in milliseconds.).

**Comment by lucasponce** [comment]
> (i.e. if the axis is labeled as "ms", all values should be in milliseconds.).

Yes, that's one of the tradeoffs, as I think it's converted automatically, and in some cases, it can "push" the charts.

I'm reviewing it, as I remember I updated it in the last work in that area.

**Comment by lucasponce** [comment]
I don't think this one would have a workaround that would satisfy all cases.

In short, the charts use the d3-format to adjust the y-axis and try to show compact (as we have a dashboard of 4x3 elements).

When you have data that suits better in one or another, the "formatter" function would adjust it [1].

In the past, we brought the unit in every y-axis value, but we put the unit only at the top, with the situation where there are multiple units, the latest is for the last value.

This limitation has a workaround with the tooltip 

![image](https://user-images.githubusercontent.com/1662329/160438375-6e79baeb-f789-4c70-87a2-45284141adca.png)
![image](https://user-images.githubusercontent.com/1662329/160438418-f4ee36e6-b300-4aba-8024-a06a30c46dfc.png)

That will help the user to know which unit for a particular point.

But I don't think it's good to revert this, as we will have a very long "y-axis" that would introduce pressure in those charts with small screens.

To fix this it should change the "formatter" in [1] and find something that works well, but the corner-cases will have edges that will work worse than the current layout.

If you'd like to give it a try the [1] and check a change of the "formatter" that may work, I can give it a try, but I think that with the tooltips users shouldn't have a problem with the SI units.


[1] https://github.com/kiali/kiali/blob/master/frontend/src/utils/Formatter.ts#L97

**Comment by lucasponce** [comment]
I'm going to close it as a "not a bug".
We can re-open if when we update some base components we can implement a different design.
